packages/teuchos/comm/test/TimeMonitor/TimeMonitor_Issue3868.cpp

```diff
@@ -0,0 +1,43 @@
+// @HEADER
+// @HEADER
+
+#include "Teuchos_UnitTestHarness.hpp"
+#include "Teuchos_TimeMonitor.hpp"
+#include "Teuchos_DefaultComm.hpp"
+#include "Teuchos_CommHelpers.hpp"
+#ifdef HAVE_TEUCHOS_MPI
+#  include "Teuchos_DefaultMpiComm.hpp"
+#endif // HAVE_TEUCHOS_MPI
+
+TEUCHOS_UNIT_TEST(TimeMonitor, Issue3868) {
+  using Teuchos::TimeMonitor;
+  using Teuchos::Comm;
+  using Teuchos::outArg;
+  using Teuchos::RCP;
+  using Teuchos::REDUCE_MAX;
+  using Teuchos::reduceAll;
+
+  RCP<const Comm<int> > comm = Teuchos::DefaultComm<int>::getComm ();
+
+  const int numProcs = comm->getSize ();
+  TEST_ASSERT( numProcs > 1 );
+  if (numProcs < 4) {
+    out << "This test requires at least 4 MPI processes." << std::endl;
```

Comment:
In the future, we should just make the test fail if it has the wrong number of processes.  I'm happy to have the test for now :D